src/main/java/players/mcts/MCTSPlayer.java

```diff
@@ -247,8 +247,15 @@ protected void createRootNode(AbstractGameState gameState) {
         if (newRoot == null) {
             if (getParameters().opponentTreePolicy == MultiTree)
                 root = new MultiTreeNode(this, gameState, rnd);
-            else
-                root = SingleTreeNode.createRootNode(this, gameState, rnd, getFactory());
+            else {
+                if(getParameters().information == MCTSEnums.Information.Perfect_Information)
+                {
+                    root = new PISingleTreeNode(this, gameState, rnd);
+                }
+                else {
```

Comment:
A better form might be:
root = ForestNode.createRootNode(...) to keep the same pattern as for SingleTree node a couple of lines below [and then I can go an refactor MultiTreeNode to the same pattern]